src/main/java/co/cask/http/RequestRouter.java

```diff
@@ -133,7 +133,7 @@ private boolean handleRequest(HttpRequest httpRequest, Channel channel, ChannelH
 
   public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) {
 
-    LOG.error("Exception caught in channel processing.", e.getCause());
+    LOG.trace("Exception caught in channel processing.", e.getCause());
```

Comment:
seems that this applies to all exceptions? Including OOM or other serious internal errors. Do we really want to disable the logging for all of those?

I think the right fix is to change the method that interprets the URL (getDestinationMethod) to throw different exceptions in different cases. One for 400 bad requests, such as malformed URLs, one for requests that can't be routed (404 not found), and one for internal errors (500). In the last case, we should log an error and return 500, in the other cases, we should only log with Trace.